lib/core/src/escape_utilities.cpp

```diff
@@ -0,0 +1,11 @@
+#include "irods/escape_utilities.hpp"
+
+#include <boost/algorithm/string/replace.hpp>
+
+namespace irods
+{
+    auto single_quotes_to_hex(const std::string& _s) -> std::string
+    {
+        return boost::replace_all_copy(_s, "'", "\\x27");
```

Comment:
No, the boost function can still throw (e.g. on allocation of a copy of `_s`).

The exception guarantee means the program isn't left in a bad state. `_s` will remain valid, which doesn't matter in our case because the replace function call didn't succeed.

Reply:
Done